File: internal/model/card.go
package model

import "time"

type GameCard struct {
	ID          int64     `db:"id" json:"id"`
	UserID      int64     `db:"user_id" json:"user_id"`
	GameID      string    `db:"game_id" json:"game_id"`
	Rank        string    `db:"rank" json:"rank"`
	Role        string    `db:"role" json:"role"`
	Description string    `db:"description" json:"description"`
	IsActive    bool      `db:"is_active" json:"is_active"`
	CreatedAt   time.Time `db:"created_at" json:"created_at"`
	UpdatedAt   time.Time `db:"updated_at" json:"updated_at"`
}

type FeedCard struct {
	GameCard
	Username string `db:"username" json:"username"`
	Age      int    `db:"age" json:"age"`
	Language string `db:"language" json:"language"`
	Region   string `db:"region" json:"region"`
	Discord  string `db:"discord" json:"discord,omitempty"`
	Telegram string `db:"telegram" json:"telegram,omitempty"`
}

File: frontend/src/types/card.ts
export type GameID = "valorant" | "cs2" | "lol" | "dota2";
export type Rank =
    | "Iron" | "Bronze" | "Silver" | "Gold"
    | "Platinum" | "Diamond" | "Ascendant" | "Immortal" | "Radiant";
export type Role = "Duelist" | "Initiator" | "Controller" | "Sentinel";

export interface GameCard {
    id: number;
    user_id: number;
    game_id: GameID;
    rank: Rank;
    role: Role;
    description: string;
    is_active: boolean;
    created_at: string;
    updated_at: string;
}

export interface FeedCard extends GameCard {
    username: string;
    age: number;
    language: string;
    region: string;
    discord?: string;
    telegram?: string;
    avatar_url?: string;
    bio?: string;
}

export interface CreateCardInput {
    game_id: GameID;
    rank: Rank;
    role: Role;
    description: string;
}

export interface FeedFilters {
    rank_min?: Rank;
    rank_max?: Rank;
    role?: Role;
    language?: string;
    region?: string;
    age_min?: number;
    age_max?: number;
}